.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,69 @@
+name: E2E Test On PR
+
+on:
+  workflow_run:
+    workflows: ["Bump versions"]
+    types:
+      - completed
+  push:
+  pull_request:
+   types: [opened, synchronize, reopened, closed]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    env:
+      CI: true
+      NODE_ENV: test
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    
+    steps:
+      - name: Log node version
+        run: node --version
+
+      - uses: actions/checkout@v2
+        with:
+         fetch-depth: '0'
+      
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Run tsc
+        run: yarn tsc
+        
+      - name: Check updates in dependencies
+        run: yarn backstage-cli versions:bump
```

Comment:
I think I would remove this from this step or we would be building something that's not what we have in the PR.

Then I'd leave this step only in the nightly workflow.

@Xantier  opinions?

Reply:
I was thinking bumping this in should be good in order to make sure cypress tests are passing for packages if we update versions

Reply:
I think I might agree with Martina. We should be able to expect that PRs are up to date with `main` and `main` should have a workflow to bump versions. Therefore this wouldn't be needed on PRs since they get correct versions numbers from `main` already

Reply:
I see, we do have this in bump-version.yml already, so it is reasonable to remove it from here then :)